assessmentModel/src/commonMain/kotlin/survey/QuestionState.kt

```diff
@@ -147,7 +157,11 @@ abstract class AbstractQuestionFieldStateImpl : QuestionFieldState {
             // `JsonNull` is used as a special placeholder for "chose not to answer".
             currentResult.jsonValue == JsonNull ||
             // Otherwise, there should be a non-null result and all the non-optional items should be selected.
-            (currentResult.jsonValue != null && itemStates.none { !it.inputItem.optional && !it.selected })
+            (currentResult.jsonValue != null && ((currentResult.jsonValue as? JsonArray)?.isNotEmpty() ?: true) && itemStates.none { !it.inputItem.optional && !it.selected })
```

Comment:
So... this will fail (I think?) to allow forward progress if the participant selects "none of the above" where the "value" is null (allowed) and therefore the selection array will be empty. Basically, an empty array is still a valid answer, right?

Reply:
Good catch. My JSON I was using for testing always had a "value"